File: 01.剑指offer/06.从尾到头打印链表/solution.py

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def reversePrint(self, head: ListNode) -> List[int]:
        res = list()

        while head != None:
            res.insert(0, head.val)
            head = head.next

        return res

File: 01.剑指offer/06.从尾到头打印链表/solution.cpp
#include <vector>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(nullptr) {}
};

class Solution {
 public:
  vector<int> reversePrint(ListNode* head) {
      int length = 0;
      ListNode *ptr = head;
      while (ptr != nullptr) {
          length++;
          ptr = ptr->next;
      }

      vector<int> ret(length);

      while (head != nullptr) {
          ret[--length] = head->val;
          head = head->next;
      }

      return ret;
  }
};

